### PR TITLE
fix shapes order

### PR DESCRIPTION
**Pull request by lieff**
I`m tried this optimization https://github.com/memononen/nanosvg/pull/81 and have garbled image on complex svg which requires correct draw order (like tiger.svg). So, I rewrite it to tail caching and it works for me.

### CODE REVIEW

File: src/nanosvg.h [hunk 1]
```diff
@@ -435,6 +435,7 @@ typedef struct NSVGparser
 	NSVGpath* plist;
 	NSVGimage* image;
 	NSVGgradientData* gradients;
+	NSVGshape *shapesTail;
```
Comment (memononen) [inline src/nanosvg.h]: Follow the coding style:
```
NSVGshape* shapesTail;
```

File: src/nanosvg.h [hunk 2]
```diff
@@ -996,9 +997,13 @@ static void nsvg__addShape(NSVGparser* p)
 	// Set flags
 	shape->flags = (attr->visible ? NSVG_FLAGS_VISIBLE : 0x00);
 
-	// Add to head due to performance, reverse list later
-	shape->next = p->image->shapes;
-	p->image->shapes = shape;
+	// Add to tail
+	if (NULL == p->image->shapes) {
```
Comment (memononen) [inline src/nanosvg.h]: Follow the coding style, no need for braces.
```
if (p->image->shapes == NULL)
```

**Comment by lieff** [comment]
Fixed. NULL == p->image->shapes is for reliability, NULL = p->image->shapes typo wont compile.

**Comment by memononen** [comment]
Thanks for the fix!

I'm familiar with the type of comparison as you wrote, and I agree with it's benefits. It's just not with the style of the rest of the code.

**Pull request merged**